platforms/jvm/language-java/src/integTest/groovy/org/gradle/api/tasks/compile/JavaCompileToolchainIntegrationTest.groovy

```diff
@@ -436,7 +436,7 @@ class JavaCompileToolchainIntegrationTest extends AbstractIntegrationSpec implem
      */
     @Requires(UnitTestPreconditions.Jdk9OrLater)
     def "Java deprecation messages with different JDKs"() {
-        def jdk = AvailableJavaHomes.getJdk(javaVersion)
```

Comment:
❓ this pattern occurs [multiple times](https://github.com/search?q=repo%3Agradle%2Fgradle%20AvailableJavaHomes.getJdk(javaVersion)&type=code) in the code base.
Should it be fixed there as well?

Should the `AvailableJavaHomes` method be fixed / a new one introduced to avoid this in the future?